src/lib.rs

```diff
@@ -931,18 +951,37 @@ impl LightningNode {
                 "Failed to get payment by hash",
             )?
         {
-            self.payment_from_breez_payment(breez_payment)
+            match self.movement_from_breez_payment(breez_payment)? {
+                Movement::Payment { payment } => Ok(payment),
+                Movement::ChannelClose { .. } => {
+                    permanent_failure!(
+                        "Searching a Breez payment by hash returned a channel close payment"
+                    );
+                }
+            }
         } else if let Some(invoice) = optional_invoice {
             self.payment_from_created_invoice(&invoice)
         } else {
             invalid_input!("No payment with provided hash was found");
         }
     }
 
-    fn payment_from_breez_payment(
+    fn movement_from_breez_payment(
+        &self,
+        breez_payment: breez_sdk_core::Payment,
+    ) -> Result<Movement> {
+        match &breez_payment.details {
+            PaymentDetails::Ln { .. } => self.movement_from_breez_ln_payment(breez_payment),
+            PaymentDetails::ClosedChannel { data } => {
+                self.movement_from_breez_closed_channel_payment(&breez_payment, data)
+            }
+        }
+    }
+
+    fn movement_from_breez_ln_payment(
         &self,
         breez_payment: breez_sdk_core::Payment,
-    ) -> Result<Payment> {
+    ) -> Result<Movement> {
         let payment_details = match breez_payment.details {
```

Comment:
Any reason why we don't supply the `LnPaymentDetails` as an argument, as done in `movement_from_breez_closed_channel_payments`?

Reply:
Applied in [c94b054](https://github.com/getlipa/lipa-lightning-lib/pull/868/commits/c94b054cbdd4b54ad373a221c9c2edb42ce0adc5)